backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java

```diff
@@ -98,6 +103,115 @@ public final int hashCode() {
+
+            static SSLValidationStrategy from(String rawValue) {
+                Preconditions.checkNotNull(rawValue);
+
+                return Stream.of(values())
+                    .filter(strategy -> strategy.name().equalsIgnoreCase(rawValue))
+                    .findAny()
+                    .orElseThrow(() -> new IllegalArgumentException(String.format("invalid strategy '%s'", rawValue)));
+
+            }
+        }
+
+        public static class SSLTrustStore {
+
+            public static SSLTrustStore of(String filePath, String password) {
+                return new SSLTrustStore(filePath, password);
+            }
+
+            private final String filePath;
+            private final String password;
+
+            private SSLTrustStore(String filePath, String password) {
+                Preconditions.checkNotNull(filePath, ELASTICSEARCH_HTTPS_TRUST_STORE_PATH + " cannot be null when " + ELASTICSEARCH_HTTPS_TRUST_STORE_PASSWORD + " is specified");
+                Preconditions.checkNotNull(password, ELASTICSEARCH_HTTPS_TRUST_STORE_PASSWORD + " cannot be null when " + ELASTICSEARCH_HTTPS_TRUST_STORE_PATH + " is specified");
+
+                this.filePath = filePath;
+                this.password = password;
+            }
+
+            public String getFilePath() {
+                return filePath;
+            }
+
+            public String getPassword() {
+                return password;
+            }
+
+            @Override
+            public final boolean equals(Object o) {
+                if (o instanceof SSLTrustStore) {
+                    SSLTrustStore that = (SSLTrustStore) o;
+
+                    return Objects.equals(this.filePath, that.filePath)
+                        && Objects.equals(this.password, that.password);
+                }
+                return false;
+            }
+
+            @Override
+            public final int hashCode() {
+                return Objects.hash(filePath, password);
+            }
+        }
+
+        static SSLTrustConfiguration defaultBehavior() {
+            return new SSLTrustConfiguration(SSLValidationStrategy.DEFAULT, Optional.empty());
+        }
+
+        static SSLTrustConfiguration ignore() {
+            return new SSLTrustConfiguration(SSLValidationStrategy.IGNORE, Optional.empty());
+        }
+
+        static SSLTrustConfiguration override(SSLTrustStore sslTrustStore) {
+            return new SSLTrustConfiguration(OVERRIDE, Optional.of(sslTrustStore));
+        }
+
+        private final SSLValidationStrategy strategy;
+        private final Optional<SSLTrustStore> trustStore;
+
+        private SSLTrustConfiguration(SSLValidationStrategy strategy, Optional<SSLTrustStore> trustStore) {
+            Preconditions.checkNotNull(strategy);
+            Preconditions.checkNotNull(trustStore);
+            Preconditions.checkArgument(strategy != OVERRIDE || trustStore.isPresent(), "OVERRIDE strategy requires trustStore to be present");
```

Comment:
please try to avoid negative conditions. The code here should match the message and it's obviously not the case

Reply:
actually it's. I just simplified the matching condition when `trategy == OVERRIDE && !trustStore.isPresent()` then throw. Had to negate it to be able to use Preconditions. I have test to ensure that. `ElasticSearchConfigurationTest.WhenOverride.fromPropertiesShouldThrowWhenTrustStoreIsNotProvided()`